04-store-api/starter/controllers/products.js

```diff
@@ -0,0 +1,76 @@
+const Product = require('../models/product')
+
+const getAllProductsStatic = async(req,res) => {
+    const products = await Product.find({price: {$gt:30}})
+    .sort('price')
+    .select('name price')
+    .limit(40)
+
+    res.status(200).json({products, nbHits: products.length})
+}
+
+const getAllProducts = async(req,res) => {
+    const {featured, company, name, sort, fields, numericFilters} = req.query
+    const queryObject = {}
+
+    if (featured) {
+        queryObject.featured = featured === 'true'? true : false
+    }
+    if (company) {
+        queryObject.company = company
+    }
+    if (name) {
+        queryObject.name = {$regex: name, $options: 'i'}
+    }
+    if (numericFilters) {
+        const operatorMap = {
+            '>': '$gt',
+            '>=': '$gte',
+            '=': '$eq',
+            '<': '$lt',
+            '<=': '$lte',
+        }
+        const regEx= /\b(<|>|>=|=|<|<=)\b/g;
+        let filters = numericFilters.replace(
+            regEx,
+            (match) =>`-${operatorMap[match]}-`
+        )
+        const options = ['price', 'rating'];
+        filters = filters.split(',').forEach((item) => {
+            const [field, operator, value] = item.split('-')
+            if (options.includes(field)) {
+                queryObject[field] = {[operator]: Number(value)}
+            }
+        })
+        //console.log(filters)
+    }
+    console.log(queryObject)
+
+    let result = Product.find(queryObject)
+    // Sorting the items
+    if (sort) {
+        const sortList = sort.split(',').join(' ');
+        result = result.sort(sortList)
+    }
+    else {
+        result = result.sort('createdAt')
+    } 
+    if (fields) {
+        const fieldList = fields.split(',').join(' ');
+        result = result.sort(fieldList)
```

Comment:
You need to use the select method here. Ex: result.select(fieldList)

Reply:
Alright, on it